$NetBSD: patch-CMakeLists.txt,v 1.1.22.1 2019/01/29 13:01:46 bsiegert Exp $

Split configuration between mysql-client and mysql-server.

Backport of https://github.com/mysql/mysql-server/commit/98ed3d8bc8ad724686d26c7bf98dced3bd1777be
Avoid disclosure of files from a client to a malicious server, described here:
https://gwillem.gitlab.io/2019/01/17/adminer-4.6.2-file-disclosure-vulnerability/

--- CMakeLists.txt.orig	2018-10-04 05:48:22.000000000 +0000
+++ CMakeLists.txt
@@ -408,7 +408,7 @@ IF(REPRODUCIBLE_BUILD)
 ENDIF()
 
 OPTION(ENABLED_LOCAL_INFILE
- "If we should enable LOAD DATA LOCAL by default" ${IF_WIN})
+ "If we should enable LOAD DATA LOCAL by default" OFF)
 MARK_AS_ADVANCED(ENABLED_LOCAL_INFILE)
 
 OPTION(OPTIMIZER_TRACE "Support tracing of Optimizer" ON)
@@ -636,7 +636,6 @@ ADD_SUBDIRECTORY(vio)
 ADD_SUBDIRECTORY(regex)
 ADD_SUBDIRECTORY(mysys)
 ADD_SUBDIRECTORY(mysys_ssl)
-ADD_SUBDIRECTORY(libmysql)
 ADD_SUBDIRECTORY(libbinlogevents)
 ADD_SUBDIRECTORY(libbinlogstandalone)
 
@@ -674,12 +673,12 @@ ADD_SUBDIRECTORY(client)
 ADD_SUBDIRECTORY(sql/share)
 ADD_SUBDIRECTORY(libservices)
 
-IF(UNIX)
+IF(WITHOUT_SERVER)
+  ADD_SUBDIRECTORY(libmysql)
   ADD_SUBDIRECTORY(man)
-ENDIF()
-
-IF(NOT WITHOUT_SERVER)
   ADD_SUBDIRECTORY(testclients)
+ELSE()
+  ADD_SUBDIRECTORY(libmysql)
   ADD_SUBDIRECTORY(sql)
   OPTION (WITH_EMBEDDED_SERVER "Compile MySQL with embedded server" OFF)
   IF(WITH_EMBEDDED_SERVER) 
